libtransmission/rpc-server.cc

```diff
@@ -51,12 +58,49 @@ using namespace std::literals;
 
 static char constexpr MyName[] = "RPC Server";
 
+static char constexpr TrUnixSocketPrefix[] = "unix:";
+
+/* The maximum size of a unix socket path is defined per-platform based on sockaddr_un.sun_path.
+ * On Windows the fallback is the length of the unix prefix. Subtracting one at the end is for
+ * double counting null terminators from sun_path and TrUnixSocketPrefix. */
+#ifdef _WIN32
+auto inline constexpr TrUnixAddrStrLen = size_t{ sizeof(TrUnixSocketPrefix) };
+#else
+auto inline constexpr TrUnixAddrStrLen = size_t{ sizeof(((struct sockaddr_un*)nullptr)->sun_path) + sizeof(TrUnixSocketPrefix) -
+                                                 1 };
+#endif
+
+enum tr_rpc_address_type
+{
+    TR_RPC_AF_INET,
+    TR_RPC_AF_INET6,
+    TR_RPC_AF_UNIX
+};
+
+struct tr_rpc_address
+{
+    tr_rpc_address_type type;
+    union
+    {
+        struct in6_addr;
+        struct in_addr;
+        char unixSocketPath[TrUnixAddrStrLen];
+    } addr;
+};
+
+static tr_rpc_address constexpr tr_rpc_inaddr_any = { TR_RPC_AF_INET, { INADDR_ANY } };
```

Comment:
I wonder if this generates a warning on some platforms right now given that `INADDR_ANY` expands to e.g. `(u_int32_t)0x00000000` on my Mac which doesn't fit into a `char`.

Reply:
Not quite sure what you mean here, at least in C, the first member of the union should be populated for that field, which would be `struct in_addr`, so this should always be correct. However, this is C++ so maybe that's different? I could not find anything conclusive googling.

Reply:
In what's currently committed to `main`, there's no `struct in_addr` member in the union, that's why I was suspicious ;)